src/main/java/org/peergos/Server.java

```diff
@@ -13,106 +15,209 @@
-            }
-            List<Map> mounts = config.getPropertyList("Datastore", "Spec", "mounts");
-            Optional<Map> blockMountOpt = mounts.stream().filter(m -> m.get("mountpoint").equals("/blocks")).findFirst();
-            if (blockMountOpt.isEmpty()) {
-                throw new IllegalStateException("Unable to fine /blocks mount");
-            }
-            Config blockConfig = new Config(blockMountOpt.get());
-            String blockPath = blockConfig.getProperty("child", "path");
-            String blockShardFunc = blockConfig.getProperty("child", "shardFunc");
-            String blockType = blockConfig.getProperty("child", "type");
-            if (!(blockPath.equals("blocks") && blockShardFunc.equals("/repo/flatfs/shard/v1/next-to-last/2")
-                && blockType.equals("flatfs"))) {
-                throw new IllegalStateException("Expecting flatfs mount at /blocks");
-            }
-            Path blocksPath = Path.of(System.getenv("HOME"), ".ipfs", "blocks");
-            FileBlockstore blocks = new FileBlockstore(blocksPath);
-            RecordStore records = new RamRecordStore();
-            ProviderStore providers = new RamProviderStore();
-            int hostPort = 6001; // 6001
-            HostBuilder builder = new HostBuilder().setIdentity(privKey).listenLocalhost(hostPort);
-            Multihash ourPeerId = Multihash.deserialize(builder.getPeerId().getBytes());
-            Kademlia dht = new Kademlia(new KademliaEngine(ourPeerId, providers, records), false);
-            CircuitStopProtocol.Binding stop = new CircuitStopProtocol.Binding();
-            CircuitHopProtocol.RelayManager relayManager = CircuitHopProtocol.RelayManager.limitTo(builder.getPrivateKey(), ourPeerId, 5);
-            builder = builder.addProtocols(List.of(
-                    new Ping(),
-                    new AutonatProtocol.Binding(),
-                    new CircuitHopProtocol.Binding(relayManager, stop),
-                    new Bitswap(new BitswapEngine(blocks)),
-                    dht));
-
-            Host node = builder.build();
-            node.start().join();
-            System.out.println("Node started and listening on " + node.listenAddresses());
-
-//            Multiaddr bootstrapNode = Multiaddr.fromString("/dnsaddr/bootstrap.libp2p.io/p2p/QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt");
-//            KademliaController bootstrap = builder.getWanDht().get().dial(node, bootstrapNode).getController().join();
-
-            APIService localAPI = new APIService();
-            String apiStr = config.getProperty("Addresses", "API");
-            String apiAddress = "/ip4/127.0.0.1/tcp/";
-            String apiPortStr = apiStr.startsWith(apiAddress) ? apiStr.substring(apiAddress.length()) : null;
-            if (apiPortStr == null) {
-                throw new IllegalStateException("Unable to determine API port");
+    public Server() throws Exception {
+        String home = System.getenv("HOME");
```

Comment:
Can we support he env var that ipfs uses to set the base .ipfs dir?

Reply:
done